packages/material-ui/src/styles/mergeClasses.js

```diff
@@ -0,0 +1,41 @@
+import warning from 'warning';
+import getDisplayName from 'recompose/getDisplayName';
+
+function mergeClasses(options = {}) {
+  const { baseClasses, newClasses, Component, noBase = false } = options;
+
+  if (!newClasses) {
+    return baseClasses;
+  }
+
+  return {
+    ...baseClasses,
+    ...Object.keys(newClasses).reduce((accumulator, key) => {
+      warning(
+        baseClasses[key] || noBase,
+        [
+          `Material-UI: the key \`${key}\` ` +
+            `provided to the classes property is not implemented in ${getDisplayName(Component)}.`,
+          `You can only override one of the following: ${Object.keys(baseClasses).join(',')}`,
+        ].join('\n'),
+      );
+
+      warning(
+        !newClasses[key] || typeof newClasses[key] === 'string',
+        [
+          `Material-UI: the key \`${key}\` ` +
+            `provided to the classes property is not valid for ${getDisplayName(Component)}.`,
+          `You need to provide a non empty string instead of: ${newClasses[key]}.`,
```

Comment:
@oliviertassinari  Sorry to bother you here but I think this may be an issue...
I see the condition `|| typeof newClasses[key] === 'string'` on line 24 and then this message: `...You need to provide a non empty string instead of...`
I'm confused... is this right? I'm getting this warning but it works (the class is applied... `newClasses[key]` is a non empty string).
> `root` provided to the classes property is not valid for ListItem.
> You need to provide a non empty string instead of: css-month-items-i14xvf

That's the message I get

Should this `||` be a `&&`? Or should the message be slightly different? Or... I'm completely wrong, of course...

Reply:
@oliviertassinari I found what is happening. As I suspected, it is not an issue with MUI :)
I was passing what the glamor `css` function returns (implementing my own override instead of using `withStyles` HOC), which is not a string but an object with a `toString` method on it. So it goes wrong through this validation, but when converted to string, it works. I solved by using `toString()` when passing it to MUI. So now no warning, and it works!
Sorry for the delayed response, I was digging into it and I just found the real cause. It may help for future reference if someone is trying to take a similar approach.

Reply:
It's good to now. I guess it's better to be explicit about it. I have seen too many people try crazy stuff with this API.

Reply:
👍 
In my case I'm trying to avoid using the HOC mainly because it is not friendly with the new React Context API (correct me if I'm wrong), and thus I created a standard function that process the style objects when I call it and pass the theme with a custom provider I created using this new context API with the render prop.

Reply:
> it is not friendly with the new React Context API

What does it mean?

Reply:
> What does it mean?

That it is using [this](https://reactjs.org/docs/legacy-context.html) instead of [this](https://reactjs.org/docs/context.html)

In practical terms the new theme is used in a declarative way by rendering a consumer and using the children on the default render prop.

So my component wrapper looks like this:
```
      <ThemeProvider.Consumer>
        {theme => {
          return (
            <List classes={processStyleObjects(classes, theme)}>
              {this.props.children}
            </List>
          )
        }}
      </ThemeProvider.Consumer>
```

Instead of creating the HOC.

Reply:
It's an implementation detail. It doesn't matter to end users. You can very well using recompose toRenderProps helper with withTheme HOC. Or use the 11 lines render prop wrapper we document for withStyles.

Reply:
Indeed, it is.

> You can very well using recompose toRenderProps helper with withTheme HOC

😲 Thank you for the tip! I didn't know about that helper
> Or use the 11 lines render prop wrapper we document for withStyles

Do you have a link? I can't find that example (I'm reading [here](https://material-ui.com/guides/typescript/#decorating-components))

Reply:
- https://github.com/acdlite/recompose/blob/master/src/packages/recompose/toRenderProps.js
- https://material-ui.com/customization/css-in-js/#render-props-api-11-lines-